src/amber/support/predicates.cr

```diff
@@ -0,0 +1,174 @@
+module Amber
+  module Support
+    module Predicates
+      module String
+        macro included
+
+          def str?
+            self.is_a? String
+          end
+
+          # email validation
+          def email?
+            !!self.match(/^[_]*([a-z0-9]+(\.|_*)?)+@([a-z][a-z0-9-]+(\.|-*\.))+[a-z]{2,6}$/)
+          end
+
+          # domain validation
+          def domain?
+            !!self.match(/^([a-z][a-z0-9-]+(\.|-*\.))+[a-z]{2,6}$/)
+          end
+
+          # url validation
+          def url?
+            !!self.match(/^(http(s)?(:\/\/))?(www\.)?[a-zA-Z0-9-_\.]+(\.[a-zA-Z0-9]{2,})([-a-zA-Z0-9:%_\+.~#?&\/\/=]*)/)
+          end
+
+          # ip v4 validation
+          def ipv4?
+            !!self.match(/^(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$/)
+          end
+
+          # ip v6 validation
+          def ipv6?
+            !!self.match(/^(?:4[0-9]{12}(?:[0-9]{3})?|5[1-5][0-9]{14}|6(?:011|5[0-9][0-9])[0-9]{12}|3[47][0-9]{13}|3(?:0[0-5]|[68][0-9])[0-9]{11}|(?:2131|1800|35\d{3})\d{11})$/)
+          end
+
+          # mac address validation
+          def mac_address?
+            !!self.match(/^([0-9a-fA-F][0-9a-fA-F]:){5}([0-9a-fA-F][0-9a-fA-F])$/)
+          end
+
+          # hex color validation
+          def hex_color?
+            !!self.match(/^#?([0-9A-F]{3}|[0-9A-F]{6})$/i)
+          end
+
+          # hexadecimal validation
+          def hex?
+            !!self.match(/^(0x)?[0-9A-F]+$/i)
+          end
+
+          # alpha characters validation
+          def alpha?(locale = "en-US")
+            !!self.match(ALPHA[matches.has_key?(locale) ? locale : "en-US"])
```

Comment:
from where does `matches` come from in `ALPHA[matches.has_key?(locale) ? locale : "en-US"]` ?

idea:
```cr
# somewhere outside:
DEFAULT_LOCALE = "en-US"

def alpha?(locale = DEFAULT_LOCALE)
  regex = ALPHA[locale]? || ALPHA[DEFAULT_LOCALE]
  !!self.match regex
end
```
I promise, this is the last comment 😛